### DISCUSSION TITLE
Is it possible to differentiate between different `globalDependencies` for different actions in the `pipeline`?

### DISCUSSION THREAD

**Discussion by maurice-at-someday** [vercel/turborepo#4548] [Help] [answered]
Is it possible to differentiate between different `globalDependencies` for different actions in the `pipeline`?

We have this in out `turbo.json` file:
```
  "globalDependencies": [
    "**/.env",
    "**/.env.local",
    "**/tsconfig.json"
  ],
```

And for the a few actions like `build` we have environment values in there that are injected into the build artifacts and that change between environment from `local`, `develop`, `staging` and `production`.

However there are other pipeline actions like `lint` and `type-check` that do not depend on these environment variables but only on the file in the repo. So for those actions I would like to exclude the environment variables so Turbo caches can be shared between environments.

Not hugely important as this isn't the biggest time cost in the various CI builds but would still save is some time if possible.

**Comment by anthonyshew** [comment, marked as answer]
A global dependency is meant to be global, rather than per-task. Per-task dependencies should be expressed in their task definitions.
